src/debugger/titaniumDebugConfigurationProvider.ts

```diff
@@ -2,19 +2,29 @@ import * as getPort from 'get-port';
 import * as vscode from 'vscode';
 import * as which from 'which';
 import { UserCancellation } from '../commands';
-import { WorkspaceState } from '../constants';
-import { ExtensionContainer } from '../container';
-import project from '../project';
-import { quickPick, selectDevice, selectiOSCertificate, selectiOSProvisioningProfile, selectPlatform } from '../quickpicks/common';
-import { getCorrectCertificateName, nameForTarget, targetsForPlatform } from '../utils';
-import { IosCertificateType } from '../types/common';
+import { selectPlatform } from '../quickpicks/common';
+import { AppBuildTaskDefinitionBase } from '../tasks/buildTaskProvider';
+import { addTask, getTasks } from '../tasks/tasksHelper';
+import { TitaniumTaskDefinitionBase } from '../tasks/commandTaskProvider';
+
+function validateTask (task: TitaniumTaskDefinitionBase, ourConfig: vscode.DebugConfiguration): void {
+	// TODO: investigate whether it is possible to update the task
+	if (ourConfig.platform && ourConfig.platform !== task.titaniumBuild.platform) {
+		throw new Error('Debug configuration platform and preLaunchTask platform do not match');
+	} else if (!task.problemMatcher || !task.problemMatcher.includes('$ti-app-launch')) {
+		throw new Error(`Pre-launch task "${task.label}" requires the "$ti-app-launch" problem matcher to be set`);
+	} else if (!task.isBackground) {
+		throw new Error(`Pre-launch task "${task.label}" requires "isBackground" to be set to true`);
+	}
+}
 
 export class TitaniumDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
 
 	public async resolveDebugConfiguration (folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration> {
 		if (!config.projectDir) {
 			config.projectDir = '${workspaceFolder}'; // eslint-disable-line no-template-curly-in-string
 		}
+
 		const ourConfig: vscode.DebugConfiguration = {
 			...config
```

Comment:
<!--
  0 failure: 
  1 warning:  src/debugger/tita...
  
  
  DangerID: danger-id-Danger;
  File: src/debugger/titaniumDebugConfigurationProvider.ts;
  Line: 29;
-->

- :warning: src/debugger/titaniumDebugConfigurationProvider.ts line 29 – Rest/spread properties are not supported until Node.js 8.3.0. The configured version range is '>=8.0.0'. (node/no-unsupported-features/es-syntax)